### PR TITLE
feat(recent-search): Save user recent search [SEN-351]

### PR DESCRIPTION
**Pull request by billyvg**
When a search is performed from search bar on issues stream, save the
search term for user+org

Fixes SEN-351
Is blocked by SEN-353

### CODE REVIEW

File: src/sentry/static/sentry/app/views/stream/searchBar.jsx [hunk 1]
```diff
@@ -71,13 +75,33 @@ class SearchBar extends React.Component {
     );
   };
 
+  handleSearch = query => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/stream/searchBar.jsx]: We could push this logic into `<SmartSearchBar>` when we need to support Events search too...

Comment (billyvg) [reply src/sentry/static/sentry/app/views/stream/searchBar.jsx]: I'll just go ahead and do that now.

File: tests/js/spec/views/stream/searchBar.spec.jsx [hunk 2]
```diff
@@ -19,9 +19,7 @@ describe('SearchBar', function() {
       context: {organization: {id: '123'}},
     };
 
-    tagValuePromise = new Promise(function(resolve, reject) {
-      return resolve([]);
-    });
+    tagValuePromise = Promise.resolve([]);
```
Comment (markstory) [inline tests/js/spec/views/stream/searchBar.spec.jsx]: :clap:

**Review by markstory — APPROVED** [review]

**Pull request merged**